### PR TITLE
feat(waf): add a datasource to get the list of anti crawler rules

### PR DESCRIPTION
**Pull request by ruwenqiang123**
**What this PR does / why we need it**:

Add a datasource to get the list of anti crawler rules.

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
1.support a new datasource to get the list of anti crawler rules.
2.support related document and acceptance test.
```

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

```
$ make testacc TEST="./huaweicloud/services/acceptance/waf" TESTARGS="-run TestAccDataSourceRulesAntiCrawler_basic"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/waf -v -run TestAccDataSourceRulesAntiCrawler_basic -timeout 360m -parallel 4
=== RUN   TestAccDataSourceRulesAntiCrawler_basic
=== PAUSE TestAccDataSourceRulesAntiCrawler_basic
=== CONT  TestAccDataSourceRulesAntiCrawler_basic
--- PASS: TestAccDataSourceRulesAntiCrawler_basic (502.23s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/waf       502.300s
```

### CODE REVIEW

File: docs/data-sources/waf_rules_anti_crawler.md [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+* `rule_id` - (Optional, String) Specifies the ID of the anti crawler rule.
+
+* `name` - (Optional, String) Specifies the name of the anti crawler rule.
+
+* `protection_mode` - (Optional, String) Specifies the protection mode of the anti crawler rule.
+  The valid values are as follows:
+  + **anticrawler_except_url**: All paths are protected except the one specified in the queried anti crawler rules.
+  + **anticrawler_specific_url**: The specified path is protected in the queried anti crawler rules.
+  
+  -> The API default query the anti crawler rules in **anticrawler_except_url** protection mode.
```
Comment (jinyangyang222) [inline docs/data-sources/waf_rules_anti_crawler.md:43]:
```suggestion
  -> If omitted, the API default query the anti crawler rules in **anticrawler_except_url** protection mode.
```

Comment (ruwenqiang123) [reply docs/data-sources/waf_rules_anti_crawler.md]: done

**Pull request merged**